python/ray/tune/ray_trial_executor.py

```diff
@@ -301,18 +302,6 @@ def _setup_remote_runner(self, trial):
                     "implemented and return True.")
             return existing_runner
 
```

Comment:
Nit: Can we add an

```
assert len(self._cached_actor_pg) == 0 or not self._reuse_actors
```

here just to be sure?

Reply:
I like to throw assertion in when the assumption is not obvious. However, in this case, I am not sure if I understand the intention of an assertion. 
`assert len(self._cached_actor_pg) == 0` alone already gives `True`, as the previous condition check was `len(self._cached_actor_pg) > 0` and always returns in previous conditional branch.

btw, I took another look at the code. There does seem to have some logic saying that if there is no pending trial of that pgf, then we shouldn't cache the actor anymore. And we will destroy the actor: https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/tune/ray_trial_executor.py?L521

I am not sure why this logic wasn't kicked in in my case (i.e. I have outstanding cached actor when the Tune experiment finishes)